apps/condo/domains/common/components/GraphQlSearchInput/GraphQlSearchInput.tsx

```diff
@@ -202,29 +224,45 @@ export const GraphQlSearchInput: React.FC<ISearchInputProps> = (props) => {
     )
 
     const handleScroll = useCallback(async (scrollEvent) => {
-        if (isNeedToLoadNewElements(scrollEvent, isLoading)) {
-            await throttledSearchMore(value, data.length)
+        if (isNeedToLoadNewElements(scrollEvent, isSearchLoading)) {
+            await throttledSearchMore(searchValue, searchValue ? searchData.length : allData.length)
+        }
+    }, [isSearchLoading, throttledSearchMore, searchValue, searchData.length, allData.length])
+
+    useEffect(() => {
+        searchMoreSuggestions('', 0)
+        loadInitialOptions()
```

Comment:
await ?

Reply:
Effect callbacks are synchronous to prevent race conditions.

You can do something similar, but is it really necessary?
![Снимок экрана 2022-10-21 в 15 14 18](https://user-images.githubusercontent.com/56914444/197172550-3d25a052-7ad3-43f8-a5a0-b61d864beed6.png)